sympy/integrals/risch.py

```diff
@@ -643,6 +640,34 @@ def reset(self):
         self.Tfuncs = []
         self.newf = self.f
 
+    def ext_k(self, extension):
+        """
+        Args:
+            extension (str): represents a valid extension type.
+
+        Returns:
+            list: A list of indices of 'exts' where extension of
+                  type 'extension' is present.
+
+        Examples
+        ========
+
+        >>> from sympy.integrals.risch import DifferentialExtension
+        >>> from sympy import log, exp
+        >>> from sympy.abc import x
+        >>> DE = DifferentialExtension(log(x) + exp(x), x, handle_first='exp')
+        >>> DE.ext_k('log')
+        [2]
+        >>> DE.ext_k('exp')
+        [1]
+
+        """
+        valid_ext = {None, 'log', 'exp', 'tan', 'atan', 'alg'}
+        if extension not in valid_ext:
+            raise ValueError("expected a valid extension type got "
```

Comment:
I'm not sure that raising an error is worthwhile. The function is intended for internal use and, in any case, probably no harm would follow if an empty list is returned for invalid exts.